iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -423,11 +455,18 @@ bool PoshRuntimeImpl::offerService(const capro::ServiceDescription& serviceDescr
     return false;
 }
 
-void PoshRuntimeImpl::stopOfferService(const capro::ServiceDescription& serviceDescription) noexcept
+bool PoshRuntimeImpl::stopOfferService(const capro::ServiceDescription& serviceDescription) noexcept
 {
-    capro::CaproMessage msg(
-        capro::CaproMessageType::STOP_OFFER, serviceDescription, capro::CaproMessageSubType::SERVICE);
-    m_applicationPort.dispatchCaProMessage(msg);
+    if (serviceDescription.isValid())
```

Comment:
See [here](https://github.com/eclipse-iceoryx/iceoryx/pull/854#discussion_r658085718)

Reply:
see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/854#discussion_r660455240) ;)

Reply:
Fixed

Reply:
Did you maybe forget to push the changes or do I need some glasses :laughing:

Reply:
Seems like I need some new :eyeglasses: :smile: This method will be moved to the new user-facing `ServiceDiscovery` API and therefore shall not just terminate in case of wrong user input, but instead inform the user on what went wrong. Hence, no change here.